.travis.yml

```diff
@@ -1,70 +1,57 @@
+      before_install: ln -s /usr/local/bin/pip3 /usr/local/bin/pip
 
-before_install:
+install:
   # OS and default Python info
-  - uname -a
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then sw_vers; fi
-  # Linux packages needed for Qt to work.
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get update; fi
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install libxkbcommon-x11-0; fi
-  # Python 3 installation required
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then bash package/install_osx.sh; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then pyenv install 3.6.5; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then pyenv versions; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then pyenv global 3.6.5; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then eval "$(pyenv init -)"; fi
-  - pip install --upgrade pip setuptools
+  - uname -a ; python3 --version ; pip3 --version ; sw_vers || true
+  - pip3 install --upgrade pip setuptools
   # Check everything was correctly installed
   - echo $PATH
-  - python --version
-  - python -c "import struct; print(struct.calcsize('P') * 8)"
-  - python -c "import sys; print(sys.executable)"
-  - python -m pip --version
-  - pip --version
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then virtualenv venv --python=python3; fi
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then source venv/bin/activate; fi
-
-install:
+  - python3 -c "import struct; print(struct.calcsize('P') * 8)"
+  - python3 -c "import sys; print(sys.executable)"
   # Install Mu and its dependencies
-  - pip install .[dev]
+  - pip3 install .[dev]
   # Check everything was correctly installed
-  - pip freeze
+  - pip3 freeze
 
 script:
   # For Linux run only the tests on 3.5 and the full checker elsewhere, all need the "X Virtual Framebuffer"
-  - if [ "$TRAVIS_OS_NAME" = "linux" ] && [ "$TRAVIS_PYTHON_VERSION" = "3.5" ]; then xvfb-run make coverage; fi
-  - if [ "$TRAVIS_OS_NAME" = "linux" ] && [ "$TRAVIS_PYTHON_VERSION" != "3.5" ]; then xvfb-run make check; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ] && [ "$TRAVIS_PYTHON_VERSION" = "3.5" ]; then make coverage; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ] && [ "$TRAVIS_PYTHON_VERSION" != "3.5" ]; then make check; fi
   - make clean
-
   # Run the tests on macOS and package it: "make macos" runs checks+tests first.
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then make macos; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then mkdir dist; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then zip -r -X dist/mu-editor.zip macOS/mu-editor.app; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then du -sk dist/; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then mv dist/mu-editor.zip dist/mu-editor_$(date '+%Y-%m-%d_%H_%M')_${TRAVIS_BRANCH}_${TRAVIS_COMMIT:0:7}.zip; fi
+  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then
+      make macos;
+      mkdir dist;
+      zip -r -X dist/mu-editor.zip macOS/mu-editor.app;
+      du -sk dist/;
+      mv dist/mu-editor.zip dist/mu-editor_$(date '+%Y-%m-%d_%H_%M')_${TRAVIS_BRANCH}_${TRAVIS_COMMIT:0:7}.zip;
+    fi
```

Comment:
One advantage of having these as individual entries, is that Travis lets you expand/collapse them, and even when it doesn't it's easy to see the individual steps as they are separated by green text with the commands.

As an example, if I wanted to see the output of `du -sk dist` I have to read the thousands of lines outputted by this single entry, as even if I search for the text `du -sk dist` I wont' find the specific output entry.